app/code/Magento/Catalog/Ui/DataProvider/Product/Form/Modifier/Categories.php

```diff
@@ -310,12 +310,16 @@ protected function customizeCategoriesField(array $meta)
      */
     protected function getCategoriesTree($filter = null)
     {
-        $categoryTree = $this->getCacheManager()->load(self::CATEGORY_TREE_ID . '_' . $filter);
+        $storeId = $this->locator->getStore()->getId();
+        $cacheKey = ($storeId)
+            ? self::CATEGORY_TREE_ID . '_' . $storeId . '_' . $filter
+            : self::CATEGORY_TREE_ID . '_' . $filter;
```

Comment:
```suggestion
            $cacheKey = self::CATEGORY_TREE_ID . '_' . $this->locator->getStore()->getId() . '_' . $filter;
```

Same logic for any `$storeId`.

Reply:
I can accept your suggestion *( it will work )*, but in your version of code in case we don't have ID of store cache key will look like `CATALOG_PRODUCT_CATEGORY_TREE__$filter` *( double underscore )*

@orlangur : Is it OK for you ? *( doubled underscore ? )*

Reply:
@ekalinak surely, I expected that it is `0` or `null`.

Reply:
Accepted.

Reply:
Oh wait. Your code is not good :/ You've created variable in `else` part of `if` @orlangur : How can I revert your change now ?

Reply:
Yep, it does not allow to propose suggestion for multiple lines.

Just change it with additional commit and squash into one.

Reply:
Everything done. Tests were't successfull. I can't figure it out what can be possibly wrong. @orlangur : Can you ?